File: 7-Grafos/2419-Costa-3/2419.py
repeat = input().split(' ')
matriz=[]
for i in range(int(repeat[0])):
    matriz.append(input())
cont=0
for y in range(len(matriz)):#[y]
    for x in range(len(matriz[i])):#[x]
        thebol=True
        if matriz[y][x] == '#':
            if y != 0:
                if matriz[y-1][x]=='.':
                    thebol=False
                    cont+=1
            else:
                if thebol:
                    thebol=False
                    cont+=1
            if y != int(repeat[0])-1 and thebol:
                if matriz[y+1][x]=='.':
                    thebol=False
                    cont+=1
            else:
                if thebol:
                    thebol=False
                    cont+=1
            if x != 0 and thebol:
                if matriz[y][x-1]=='.':
                    thebol=False
                    cont+=1
            else:
                if thebol:
                    thebol=False
                    cont+=1
            if x != int(repeat[1])-1 and thebol:
                if matriz[y][x+1]=='.':
                    thebol=False
                    cont+=1
            else:
                if thebol:
                    thebol=False
                    cont+=1
print(cont)

File: 7-Grafos/2419-Costa-3/2419.cs
// Online C# Editor for free
// Write, Edit and Run your C# code using C# Online Compiler

using System;

public class HelloWorld
{
    public static void Main(string[] args)
    {
        bool thebol;
        int cont=0;
        string[] repeat = Console.ReadLine().Split(' ');
        
        string[] matriz = new string [int.Parse(repeat[0])];
        
        for (int i = 0; i<int.Parse(repeat[0]); i++){
            matriz[i]=Console.ReadLine();
        }
        
        for (int y = 0; y<matriz.Length; y++){
            for (int x = 0 ; x<matriz[y].Length; x++){
                thebol=true;
                if (matriz[y][x] == '#'){
                    if (y != 0){
                        if (matriz[y-1][x]=='.'){
                            thebol=false;
                            cont++;
                            }
                        }
                    
                    else{
                        if (thebol){
                            thebol=false;
                            cont++;
                        }
                    }
                    if (y != int.Parse(repeat[0])-1 & thebol){
                        if (matriz[y+1][x]=='.'){
                            thebol=false;
                            cont++;
                        }
                    }
                    else{
                        if (thebol){
                            thebol=false;
                            cont++;
                        }
                    }
                    if (x != 0 & thebol){
                        if (matriz[y][x-1]=='.'){
                            thebol=false;
                            cont++;
                        }
                    }
                    else{
                        if (thebol){
                            thebol=false;
                            cont++;
                        }
                    }
                    if (x != int.Parse(repeat[1])-1 & thebol){
                        if (matriz[y][x+1]=='.'){
                            thebol=false;
                            cont++;
                        }
                    }
                    else{
                        if (thebol){
                            thebol=false;
                            cont++;
                        }
                    }
                }
                
            }
        }
        Console.WriteLine(cont);
        
    }
}